solo-chains/node/tanssi-relay-service/src/lib.rs

```diff
@@ -14,6 +14,602 @@
 
 pub mod chain_spec;
+
+pub type FullBackend = service::TFullBackend<Block>;
+
+pub type FullClient = service::TFullClient<
+    Block,
+    RuntimeApi,
+    WasmExecutor<(
+        sp_io::SubstrateHostFunctions,
+        frame_benchmarking::benchmarking::HostFunctions,
+    )>,
+>;
+
+pub struct NewFull {
+    pub task_manager: TaskManager,
+    pub client: Arc<FullClient>,
+    pub overseer_handle: Option<Handle>,
+    pub network: Arc<dyn sc_network::service::traits::NetworkService>,
+    pub sync_service: Arc<sc_network_sync::SyncingService<Block>>,
+    pub rpc_handlers: RpcHandlers,
+    pub backend: Arc<FullBackend>,
+}
+
+/// Custom Deps for dev Rpc extension
+struct DevDeps<C, P, B> {
+    /// The client instance to use.
+    pub client: Arc<C>,
+    /// Transaction pool instance.
+    pub pool: Arc<P>,
+    /// A copy of the chain spec.
+    pub chain_spec: Box<dyn sc_chain_spec::ChainSpec>,
+    /// Whether to deny unsafe calls
+    pub deny_unsafe: DenyUnsafe,
+    /// Backend used by the node.
+    pub backend: Arc<B>,
+    /// Manual seal command sink
+    pub command_sink: Option<futures::channel::mpsc::Sender<EngineCommand<Hash>>>,
+}
+
+fn create_rpc_extension<C, P, B>(
+    DevDeps {
+        client,
+        pool,
+        chain_spec,
+        deny_unsafe,
+        backend,
+        command_sink: maybe_command_sink,
+    }: DevDeps<C, P, B>,
+) -> Result<RpcExtension, Box<dyn std::error::Error + Send + Sync>>
+where
+    C: ProvideRuntimeApi<Block>
+        + HeaderBackend<Block>
+        + AuxStore
+        + HeaderMetadata<Block, Error = sp_blockchain::Error>
+        + Send
+        + Sync
+        + 'static,
+    C::Api: substrate_frame_rpc_system::AccountNonceApi<Block, AccountId, Nonce>,
+    C::Api: mmr_rpc::MmrRuntimeApi<Block, <Block as sp_runtime::traits::Block>::Hash, BlockNumber>,
+    C::Api: pallet_transaction_payment_rpc::TransactionPaymentRuntimeApi<Block, Balance>,
+    C::Api: BlockBuilder<Block>,
+    P: TransactionPool + Sync + Send + 'static,
+    B: Backend<Block> + Send + Sync + 'static,
+    B::State: sc_client_api::StateBackend<sp_runtime::traits::HashingFor<Block>>,
+{
+    use mmr_rpc::{Mmr, MmrApiServer};
+    use pallet_transaction_payment_rpc::{TransactionPayment, TransactionPaymentApiServer};
+    use sc_rpc_spec_v2::chain_spec::{ChainSpec, ChainSpecApiServer};
+    use substrate_frame_rpc_system::{System, SystemApiServer};
+    use substrate_state_trie_migration_rpc::{StateMigration, StateMigrationApiServer};
+
+    let mut io = RpcModule::new(());
+
+    let chain_name = chain_spec.name().to_string();
+    let genesis_hash = client
+        .hash(0)
+        .ok()
+        .flatten()
+        .expect("Genesis block exists; qed");
+    let properties = chain_spec.properties();
+
+    io.merge(ChainSpec::new(chain_name, genesis_hash, properties).into_rpc())?;
+    io.merge(StateMigration::new(client.clone(), backend.clone(), deny_unsafe).into_rpc())?;
```

Comment:
You can probably remove this because we removed the state migration pallet from starlight. And the Mmr from below is probably also not needed? If so you can remove the `mmr-rpc` dependency.

```suggestion
```